### PR TITLE
fix: update regex to allow whitespace & `+` in Windows path

### PR DESCRIPTION
**Pull request by MauriceChocoSwiss**
Updated regex to authorize whitespace in file path.

Many projects has a whitespace in folder name (like me). Many time it doesn't affect anything but here it prevent file to open.

Let me know !

### CODE REVIEW

**Comment by dominikg** [comment]
Hey, 

would you be interested in adding `+` as well? Thats used by sveltekit ( see https://github.com/yyx990803/launch-editor/issues/50#issuecomment-2227992963 ) and https://github.com/sveltejs/vite-plugin-svelte/issues/943

cc @sodatea  this would be the "expand allow list" i mentioned, might be a stop-gap solution until you find the time to implement propper escaping.

**Comment by MauriceChocoSwiss** [comment]
Why not, I didn't use sveltkit but I think it can be good for sveltkit user.

File: packages/launch-editor/index.js [hunk 1]
```diff
@@ -107,7 +107,7 @@ function launchEditor (file, specifiedEditor, onErrorCallback) {
   // https://github.com/facebook/create-react-app/pull/5431)
 
   // Allows alphanumeric characters, periods, dashes, slashes, and underscores.
```
Comment (dominikg) [inline packages/launch-editor/index.js:109]:
```suggestion
  // Allows alphanumeric characters, periods, dashes, slashes, underscores, plus and space.
```

File: packages/launch-editor/index.js [hunk 2]
```diff
@@ -107,7 +107,7 @@ function launchEditor (file, specifiedEditor, onErrorCallback) {
...
-  const WINDOWS_CMD_SAFE_FILE_NAME_PATTERN = /^([A-Za-z]:[/\\])?[\p{L}0-9/.\-_\\]+$/u
+  const WINDOWS_CMD_SAFE_FILE_NAME_PATTERN = /^([A-Za-z]:[/\\])?[\p{L}0-9/.\-_\\\s]+$/u
```
Comment (dominikg) [inline packages/launch-editor/index.js:110]:
```suggestion
  const WINDOWS_CMD_SAFE_FILE_NAME_PATTERN = /^([A-Za-z]:[/\\])?[\p{L}0-9/.\-+_\\\s]+$/u
```

**Comment by dominikg** [comment]
i wonder if you want `\s` or an actual space char in that group

**Comment by MauriceChocoSwiss** [comment]
Idon't whitch one is better, by default I use \s. I don't know if "space" char work on all platform and all situation but \s yes.

**Comment by dominikg** [comment]
is it possible to have line breaks or tabs in a file path on windows? (that file system never fails to impress, just not sure if in a positive way)

**Comment by MauriceChocoSwiss** [comment]
I really don't know... You prefer to use only spaces ?

**Comment by dominikg** [comment]
unless they are possible/allowed, i would prefer just space, but maybe @sodatea  has a different opinion. 

It would be good to get this merged soonish so we can include it in vite 5.5.

**Review by sodatea — APPROVED** [review]

**Comment by sodatea** [comment]
According to https://learn.microsoft.com/en-us/windows/win32/fileio/naming-a-file CR, LF and TABs are not allowed in a file path. So I think a single space character should suffice.

**Comment by sodatea** [comment]
As for other special characters, there seem to be many more to add for compatibility with all the popular filesystem-based routing solutions, but I'll leave that to a later PR.

**Comment by oneezy** [comment]
> As for other special characters, there seem to be many more to add for compatibility with all the popular filesystem-based routing solutions, but I'll leave that to a later PR.

yeahh it's still an issue on Win11

### Reference
**issue**: https://github.com/sveltejs/vite-plugin-svelte/issues/943#issuecomment-2308578599

**Comment by Loizzus** [comment]
I don't think banning characters is the solution to this problem. Even if you did find all the characters that are currently used by every system that depends on Vite why prevent some creative new use of valid special characters in the future? This entire feature seems very short sighted.

**Pull request merged**